extmod/os_dupterm.c

```diff
@@ -148,29 +148,42 @@ int mp_os_dupterm_rx_chr(void) {
     return -1;
 }
 
-void mp_os_dupterm_tx_strn(const char *str, size_t len) {
+int mp_os_dupterm_tx_strn(const char *str, size_t len) {
+    // Returns the minimum successful write length, or -1 if no write is attempted.
+    int ret = len, did_write = 0;
```

Comment:
This is a nitpick, but good to use `bool` for things that are booleans

Reply:
oops, missed that one